commons-app/apps-android-commons
# Fixed default license selector in Settings

**Description (required)**

Fixes #2072 Fix default license selector in Settings 

What changes did you make and why?

commons/upload/UploadPresenter.java
-* added dependency injection for SharedPreferences
-* this allows updateLicenses() to retrieve the current default license (set by user in settings) and subsequently update the UploadModel 

commons/upload/UploadModel.java
-* changed constructor so that 'String license' is set to the current default license (set by user in settings) 

**Tests performed (required)**

1.  Open app
2.  Navigate to settings
3.  Change the license setting (to preferred license)
4.  Navigate to Home and upload photo, select photo, go through wizard
5.  At step 3 of 3, check that license is automatically set to the preferred license
6.  Navigate backwards to home, then navigate to settings
7.  Change the license setting again (to new preferred license)
8.  Navigate to Home and upload photo, select photo, go through wizard
9.  At step 3 of 3, check that license is automatically set to the new preferred license
10.  Submit and check that license is as selected in photo attributes

Tested commons-app/apps-android-commons branch: master on Nexus 5 with API level 27.
